Files/Helpers/ExtensionManager.cs

```diff
@@ -0,0 +1,531 @@
+        /// </summary>
+        /// <returns></returns>
+        public async Task MarkAsLoaded()
+        {
+            // make sure package is OK to load
+            if (!AppExtension.Package.Status.VerifyIsOK())
+            {
+                return;
+            }
+
+            Enabled = true;
+
+            // Don't reload
+            if (Loaded)
+            {
+                return;
+            }
+
+            // The public folder is shared between the extension and the host.
+            // We don't use it in this sample but you can see https://github.com/Microsoft/Build2016-B808-AppExtensibilitySample for an example of it can be used.
+            StorageFolder folder = await AppExtension.GetPublicFolderAsync();
+            PublicFolderPath = folder.Path;
+            try
+            {
+                var file = await folder.GetFileAsync("FileExtensions.json");
+                var text = await FileIO.ReadTextAsync(file);
+                FileExtensions = JsonConvert.DeserializeObject<List<string>>(text);
+            } catch
+            {
+                Debug.WriteLine("Unable to get extensions");
+            }
+
+            Loaded = true;
+            Visible = Visibility.Visible;
+            RaisePropertyChanged("Visible");
+            Offline = false;
+        }
+
+        /// <summary>
+        /// Enable the extension for use
+        /// </summary>
+        /// <returns></returns>
+        public async Task Enable()
+        {
+            Enabled = true;
+            await MarkAsLoaded();
+        }
+
+        /// <summary>
+        /// Indicates to the extension manager that the extension is unloaded
+        /// </summary>
+        public void Unload()
+        {
+            // unload it
+            lock (_sync) // Calls to this functioned are queued on an await call so lock to handle one at a time
+            {
+                if (Loaded)
+                {
+                    // see if the package is offline
+                    if (!AppExtension.Package.Status.VerifyIsOK() && !AppExtension.Package.Status.PackageOffline)
+                    {
+                        Offline = true;
+                    }
+
+                    Loaded = false;
+                    Visible = Visibility.Collapsed;
+                    RaisePropertyChanged("Visible");
+                }
+            }
+        }
+
+        // user-facing action to disable the extension
+        public void Disable()
+        {
+            // only disable if it is enabled so that we don't Unload() more than once
+            if (Enabled)
+            {
+                Enabled = false;
+                Unload();
+            }
+        }
+        #region PropertyChanged
+
+        /// <summary>
+        /// Typical property changed handler so that the UI will update
+        /// </summary>
+        /// <param name="name"></param>
+        private void RaisePropertyChanged(string name)
```

Comment:
You could inherit from `ObservableObject` - it already has `OnPropertyChanged()`

Reply:
@d2dyno1 Is this change necessary? It would require rewriting a decent amount of code.